apps/api/src/routes/users/creditCards.ts

```diff
@@ -9,32 +9,66 @@ import { Hono } from "hono"
 import { z } from "zod"
 
 export const creditCardRouter = () =>
-	new Hono<{ Variables: PrivateContextVariables }>().post(
-		"/",
-		zValidator("param", z.object({ userId: idValidator })),
-		zValidator("json", creditCardValidator),
-		async (c) => {
-			const { userId } = c.req.valid("param")
-			const { prisma } = c.var
-			const data = c.req.valid("json")
+	new Hono<{ Variables: PrivateContextVariables }>()
+		.post(
+			"/",
+			zValidator("param", z.object({ userId: idValidator })),
+			zValidator("json", creditCardValidator),
+			async (c) => {
+				const { userId } = c.req.valid("param")
+				const { prisma } = c.var
+				const data = c.req.valid("json")
 
-			try {
-				const newCreditCard = await prisma.creditCard.create({
-					data: {
-						...data,
-						userId,
-					},
-				})
+				try {
+					const newCreditCard = await prisma.creditCard.create({
+						data: {
+							...data,
+							userId,
+						},
+					})
 
-				return c.json(newCreditCard, HTTP_STATUS_CODES.CREATED)
-			} catch (error) {
-				// eslint-disable-next-line no-console
-				console.error(error)
+					return c.json(newCreditCard, HTTP_STATUS_CODES.CREATED)
+				} catch (error) {
+					return c.json(
+						{ error: "Failed to create credit card" },
+						HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR,
+					)
+				}
+			},
+		)
+		.delete(
+			"/:cardId",
+			zValidator("param", z.object({ cardId: idValidator })),
+			async (c) => {
+				const { cardId } = c.req.valid("param")
+				const { prisma } = c.var
```

Comment:
destructure prisma dicrectly is nicer to look at

not ```const { prisma } = c.var```

but 
```suggestion

async ({prisma}) => {
				const { cardId } = c.req.valid("param")
				````